vpr/SRC/route/route_timing.c

```diff
@@ -1304,3 +1431,69 @@ static bool early_exit_heuristic(const t_router_opts& router_opts) {
 	return false;
 }
 
+void convert_sink_node_to_pins_of_net(const unordered_map<int,int>& node_to_pin_mapping, vector<int>& sink_nodes) {
+
+	for (size_t s = 0; s < sink_nodes.size(); ++s) {
+
+		auto mapping = node_to_pin_mapping.find(sink_nodes[s]);
+		// should always expect it find a pin mapping for its own net
+		INCREMENTAL_REROUTING_ASSERT(mapping != node_to_pin_mapping.end());

```

Comment:
seems like a check that will always be useful. +1 to Kevin's suggestion (or use if-statement + vpr_throw)